website/search/file_util.py

```diff
@@ -0,0 +1,95 @@
+import requests
+
+from website import settings
+
+
+INDEXED_TYPES = (
+    '.txt',
+    '.md',
+    '.rtf',
+    '.docx',
+    '.pdf',
+)
+
+
+def require_file_indexing(func):
+    """ Execute function only if use_file_indexing setting is true. """
+    def wrapper(*args, **kwargs):
+        if settings.USE_FILE_INDEXING:
+            return func(*args, **kwargs)
+        return None
```

Comment:
You may want to throw a logger statement in here so there won't be any "silent" failures.